dataset/models/tf/layers/conv.py

```diff
@@ -96,8 +134,71 @@ def conv_transpose(inputs, filters, kernel_size, strides, *args, **kwargs):
     return output
 
 
+def _separable_conv(transpose, inputs, filters, kernel_size, strides, padding, data_format,
+                    dilation_rate, depth_multiplier, activation, name, **kwargs):
+    dim = inputs.shape.ndims - 2
+    context = None
+    if name is not None:
+        context = tf.variable_scope(name)
+        context.__enter__()
+    if transpose:
+        conv_layer = conv_transpose
+    else:
+        conv_layer = conv
+
+    kwargs = {'kernel_size': kernel_size,
+              'strides': strides,
+              'padding': padding,
+              'data_format': data_format,
+              'dilation_rate': dilation_rate,
+              'activation': activation,
+              **kwargs}
+
+    if transpose:
+        kwargs.pop('dilation_rate')
```

Comment:
Add `dilation_rate` only when needed (not add it always and then remove sometimes)